integrations/flink/documentation.yaml

```diff
@@ -0,0 +1,79 @@
+exporter_type: included
+app_name_short: Flink
+app_name: {{app_name_short}}
+app_site_name: Flink
+app_site_url: https://flink.apache.org/
+exporter_name: the Flink exporter
+exporter_pkg_name: Flink
+exporter_repo_url: https://flink.apache.org/features/2019/03/11/prometheus-monitoring.html
+additional_prereq_info: |
+  {{app_name_short}} exposes Prometheus-format metrics when configured with
+  `metrics.reporter.prom.class: org.apache.flink.metrics.prometheus.PrometheusReporter`.
+
+   *  To determine if {{app_name_short}} is exposing metrics, run the following command:
```

Comment:
Actually, I could not get metrics to show up by using the [sample getting started manifests](https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/resource-providers/standalone/kubernetes/)... I got:

```
2023-01-09 21:26:20,313 ERROR org.apache.flink.runtime.metrics.ReporterSetup               [] - Could not instantiate metrics reporter prom. Metrics might not be exposed/reported.
java.lang.ClassNotFoundException: org.apache.flink.metrics.prometheus.PrometheusReporter
        at jdk.internal.loader.BuiltinClassLoader.loadClass(Unknown Source) ~[?:?]
        at jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(Unknown Source) ~[?:?]
        at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:?]
        at java.lang.Class.forName0(Native Method) ~[?:?]
        at java.lang.Class.forName(Unknown Source) ~[?:?]
        at org.apache.flink.runtime.metrics.ReporterSetup.loadViaReflection(ReporterSetup.java:456) ~[flink-dist-1.16.0.jar:1.16.0]
        at org.apache.flink.runtime.metrics.ReporterSetup.loadReporter(ReporterSetup.java:409) ~[flink-dist-1.16.0.jar:1.16.0]
        at org.apache.flink.runtime.metrics.ReporterSetup.setupReporters(ReporterSetup.java:328) ~[flink-dist-1.16.0.jar:1.16.0]
        at org.apache.flink.runtime.metrics.ReporterSetup.fromConfiguration(ReporterSetup.java:209) ~[flink-dist-1.16.0.jar:1.16.0]
        at org.apache.flink.runtime.entrypoint.ClusterEntrypoint.createMetricRegistry(ClusterEntrypoint.java:458) ~[flink-dist-1.16.0.jar:1.16.0]
        at org.apache.flink.runtime.entrypoint.ClusterEntrypoint.initializeServices(ClusterEntrypoint.java:397) ~[flink-dist-1.16.0.jar:1.16.0]
        at org.apache.flink.runtime.entrypoint.ClusterEntrypoint.runCluster(ClusterEntrypoint.java:282) ~[flink-dist-1.16.0.jar:1.16.0]
        at org.apache.flink.runtime.entrypoint.ClusterEntrypoint.lambda$startCluster$1(ClusterEntrypoint.java:232) ~[flink-dist-1.16.0.jar:1.16.0]
        at org.apache.flink.runtime.security.contexts.NoOpSecurityContext.runSecured(NoOpSecurityContext.java:28) [flink-dist-1.16.0.jar:1.16.0]
        at org.apache.flink.runtime.entrypoint.ClusterEntrypoint.startCluster(ClusterEntrypoint.java:229) [flink-dist-1.16.0.jar:1.16.0]
        at org.apache.flink.runtime.entrypoint.ClusterEntrypoint.runClusterEntrypoint(ClusterEntrypoint.java:729) [flink-dist-1.16.0.jar:1.16.0]
        at org.apache.flink.runtime.entrypoint.StandaloneSessionClusterEntrypoint.main(StandaloneSessionClusterEntrypoint.java:59) [flink-dist-1.16.0.jar:1.16.0]
2023-01-09 21:26:20,328 INFO  org.apache.flink.runtime.metrics.MetricRegistryImpl          [] - No metrics reporter configured, no metrics will be exposed/reported.
```

Weirdly, after referencing the OiQ yaml manifests and inspecting the diff, I got this to work by changing
```
      - name: jobmanager
        # image: apache/flink:latest
        image: apache/flink:1.15.2-scala_2.12 # fixed
        args: ["jobmanager"]
```

Did you experience the same? Also, I noticed that the OiQ yaml manifests have
```
        ports:
        ...
        - containerPort: 9249
          name: prometheus     
```

Is this necessary?

Reply:
I would expect "latest" to work, however, I am not a fan of using latest outside of a quick test environment.

With that said, I am getting metrics with our dev configs, without changes. If I remove the `prometheus` named port from the pod spec's `ports`, I can still access the port w/ port forwarding. I just loose the ability to reference the port by name, which we are not doing anywhere.

Reply:
OK, sounds good.